board.cpp

```diff
@@ -53,55 +53,153 @@ bool Board::isMovable(const QModelIndex &index)
               std::abs(index.row() - m_selectedTileIndex.row()) == 1) ) ? true : false;
 }
 
-void Board::popTiles(const QModelIndex& index)
+bool Board::popTiles(const QModelIndex& index)
 {
-    using Directions = QVector<QPoint>;
 
-    QVector<QPoint> forPopping;
-    static const Directions directions = {{-1,0}, {0,1}, {1,0}, {0,-1}};
+    auto isValid = [this] (QPoint& p)
+                    {
+                        return ( (p.x() < 0)
+                              || (p.y() < 0)
+                              || (p.x() > m_dimension - 1)
+                              || (p.y() > m_dimension - 1) ? false : true );
+                    };
+
+
     QPoint current(index.row(), index.column());
-    auto inx = [this] (const QPoint& current) -> int
-                {
-                    return (m_dimension * current.x() + current.y());
-                }; // index 2d to 1d array
+    QVector<QPoint> forPoppingH{current};
+    QVector<QPoint> forPoppingV{current};
```

Comment:
I think you can use 1 `QVector` just count the number of elements before checking and after